fix: remove pie animation id

ckifer opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description

<!--- Describe your changes in detail -->
Removing Pie `animationId` as it creates a regression from 2.x where the Pie animation pauses upon state being set in its event handlers. Removing the id doesn't cause any differences in instances where there is no custom state being set or passed

## Related Issue

<!--- This project only accepts pull requests related to open issues -->
<!--- If suggesting a new feature or change, please discuss it in an issue first -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please link to the issue here: -->
https://github.com/recharts/recharts/issues/5804

## Motivation and Context

<!--- Why is this change required? What problem does it solve? -->
solve issues before release

## How Has This Been Tested?

<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->
- test pass. See difference in storybooks
  - Current, bad - https://63da8268a0da9970db6992aa-tcfvxoomhy.chromatic.com/?path=/story/examples-pie-piewithtooltip--pie-with-tooltip

## Screenshots (if appropriate):

## Types of changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] I have added a storybook story or extended an existing story to show my changes

PavelVanecek commented:
The animationId is there to fix this thing: https://github.com/recharts/recharts/pull/5753 would you like to add a story that changes the data in a Pie and see how that animates?

How difficult would it be to not re-render the Pie on mouse over? That doesn't seem like it should be related. Why would Pie need to rerender when tooltip shows up.

ckifer commented:
Ok so correct without animationId there is no re-animation when data changes.

---

Minimally, this happens as soon as you set _any_ state inside _any_ of the event handlers, which I think it certainly a regression. Reducing that story to this 

```tsx
onMouseEnter={() => {
    setActive(true);
}}
```

Without passing `active` anywhere or doing anything else still results in the same issue. So its within Pie.

Adding something like
```ts
  Object.keys(input).forEach(key => {
    // @ts-expect-error bleh
    if (prevProps.current[key] !== input[key]) console.log('Changed', key, prevProps.current[key], input[key]);
  });

```

to `useAnimationId` results in:

```
Changed onMouseEnter
Changed children 
Changed sectors 
Changed onMouseEnter 
Changed children 
Changed sectors 
Changed onMouseEnter 
```

on any mouse enter movement whereas on initial render we only have `Changed sectors`

If I go to any `Pie` without event handlers, this doesn't happen and `sectors` change on initial render alone.

ckifer commented:
I don't really think there is a way around it - if you `setState` you are causing a re-render. Using a ref "works" but it doesn't cause anything to update and is therefore useless. We also can't tell our consumers to do that and them be ok with it even if it did work.
